### ISSUE TITLE
Command setup:config:set unusable with large media directory.

### ISSUE THREAD

**Issue by jzahedieh** [magento/magento2#8672]
### Preconditions

1. Magento EE 2.1.4, PHP 7.0.*, MySQL 5.7, Production or Developer mode.
2. Large media directory, we've currently got around 350G of product media

### Steps to reproduce

1. Run `magento setup:config:set --http-cache-hosts=host1,host2,host3`
2. Wait 3 hours for media check for `setup/src/Magento/Setup/Model/ConfigModel.php:checkInstallationFilePermissions()` to finish

---

Alternatively and quicker is to have a development machine with no media and create an invalid image folder with restrictive permissions i.e `chmod -R 440 pub/media/catalog/product/1/`

```
# magento setup:config:set --http-cache-hosts=host1,host2,host3

Overwrite the existing configuration for http-cache-hosts?[Y/n]Y
                          
  [Exception]                                          
  Missing write permissions to the following paths:    
  /usr/share/nginx/html/pub/media/catalog/product/1/5  
  /usr/share/nginx/html/pub/media/catalog/product/1    
                                                       
```

### Expected result
1. If area is not affected by command do not check write permissions
2. `magento app:config:set --http-cache-hosts=host1,host2,host3` should only check `app/etc/env.php` for writability 
3. I've run strace on it and it is doing file level check, surely directory level checks should be sufficient?

For 3 strace output.

```
lstat("/usr/share/nginx/html/pub/media/catalog/product/cache/thumbnail/88x110/beff4985b56e3afdbeabfc89641a4582/0/8/08-09-2015_denim_9.jpg", {st_mode=S_IFREG|0664, st_size=1722, ...}) = 0
stat("/usr/share/nginx/html/pub/media/catalog/product/cache/thumbnail/88x110/beff4985b56e3afdbeabfc89641a4582/0/8/08-09-2015_denim_9.jpg", {st_mode=S_IFREG|0664, st_size=1722, ...}) = 0
access("/usr/share/nginx/html/pub/media/catalog/product/cache/thumbnail/88x110/beff4985b56e3afdbeabfc89641a4582/0/8/08-09-2015_denim_9.jpg", W_OK) = 0
lstat("/usr/share/nginx/html/pub/media/catalog/product/cache/thumbnail/88x110/beff4985b56e3afdbeabfc89641a4582/0/8/08-06-2015__1.jpg", {st_mode=S_IFREG|0664, st_size=1525, ...}) = 0
stat("/usr/share/nginx/html/pub/media/catalog/product/cache/thumbnail/88x110/beff4985b56e3afdbeabfc89641a4582/0/8/08-06-2015__1.jpg", {st_mode=S_IFREG|0664, st_size=1525, ...}) = 0
access("/usr/share/nginx/html/pub/media/catalog/product/cache/thumbnail/88x110/beff4985b56e3afdbeabfc89641a4582/0/8/08-06-2015__1.jpg", W_OK) = 0
lstat("/usr/share/nginx/html/pub/media/catalog/product/cache/thumbnail/88x110/beff4985b56e3afdbeabfc89641a4582/0/8/08-07-2015__4_dl.jpg", {st_mode=S_IFREG|0664, st_size=2566, ...}) = 0
stat("/usr/share/nginx/html/pub/media/catalog/product/cache/thumbnail/88x110/beff4985b56e3afdbeabfc89641a4582/0/8/08-07-2015__4_dl.jpg", {st_mode=S_IFREG|0664, st_size=2566, ...}) = 0
access("/usr/share/nginx/html/pub/media/catalog/product/cache/thumbnail/88x110/beff4985b56e3afdbeabfc89641a4582/0/8/08-07-2015__4_dl.jpg", W_OK) = 0
lstat("/usr/share/nginx/html/pub/media/catalog/product/cache/thumbnail/88x110/beff4985b56e3afdbeabfc89641a4582/0/8/08-10-2015_silver_sh_2.jpg", {st_mode=S_IFREG|0664, st_size=1336, ...}) = 0
stat("/usr/share/nginx/html/pub/media/catalog/product/cache/thumbnail/88x110/beff4985b56e3afdbeabfc89641a4582/0/8/08-10-2015_silver_sh_2.jpg", {st_mode=S_IFREG|0664, st_size=1336, ...}) = 0
```

### Actual result
1. Performance issue or error.
2. Command doesn't update env.php with cache hosts.

### Additional
Since most of the source for these commands is in `setup/src/Magento/Setup/Console/Command/` it makes it impossible to use interceptors, plugins, preferences and DI to try and modify the behaviour in the `Magento\Setup\Model\ConfigModel:process()` command.

It would be really good if Magento could make the setup resources use the composer installed version instead of relying on a separate code base for these as it reduces the extendability.

### Related Issues

#7933
#4955

**Comment by andimov** [comment]
@jzahedieh
Thanks for reporting this issue.
Internal ticket MAGETWO-65399, which tracks this GitHub issue, is in our issue backlog.

**Comment by magento-engcom-team** [comment]
@jzahedieh, thank you for your report.
The issue is already fixed in develop branch, 2.2.0

**Comment by magento-engcom-team** [comment]
Hi @jzahedieh. 

Thank you for your report and collaboration!

The related internal Jira ticket `MC-24734` was closed as `Fixed`. 

The fix will be available with the upcoming `2.4.3` release.
